### PR TITLE
Add query_box functionality.

### PR DESCRIPTION
**Pull request by erykoff**
This was requested by @talljimbo, and adds (nest) queries for "boxes" which are regions constrained by constant lines of longitude/latitude (as opposed to polygons which have great circle boundaries).

### CODE REVIEW

File: hpgeom/healpix_geom.c [hunk 1]
```diff
@@ -1410,3 +1411,124 @@ void query_ellipse(healpix_info *hpx, double ptg_theta, double ptg_phi, double s
         }
     }
 }
+
+void query_box(healpix_info *hpx, double ptg_theta0, double ptg_theta1, double ptg_phi0,
+               double ptg_phi1, bool full_lon, int fact, struct i64rangeset *pixset,
+               int *status, char *err) {
+    if (hpx->scheme == RING) {
+        snprintf(err, ERR_SIZE, "query_ellipse only supports nest ordering.");
```
Comment (wmwv) [inline hpgeom/healpix_geom.c]: "query_box only supports nest ordering."

**Review by wmwv — CHANGES_REQUESTED** [review]
Code looks good.
Add `query_box` example to documentation.
Consider adding `reorder` to documentation if you want people to use it (you may not).

**Comment by erykoff** [comment]
I fixed the error message, and I filed #19 as a reminder to flesh out the documentation with usage examples.

**Review by wmwv — APPROVED** [review]
LGTM

**Pull request merged**